### ISSUE TITLE
hazelcast management server sometime goes Hang

### ISSUE THREAD

**Issue by ppsnumi** [hazelcast/hazelcast#16121]
Hi~

I use hazeclast(3.12.3) and hazelcast-management server(3.12.5)
but hazelcast-management server sometime hang

hzmng(management server) process live but web page not open.
jetty acceptor thread blocked in thread dump.

```
"qtp25126016-76-acceptor-0@16a3b15e-WebApp@2ef5e5e3{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}" #76 prio=3 os_prio=64 tid=0x0000000106d08800 nid=0x63 waiting for monitor entry [0xffffffff238fe000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:233)
	- waiting to lock <0x00000003c42b57c0> (a java.lang.Object)
	at org.eclipse.jetty.server.ServerConnector.accept(ServerConnector.java:385)
	at org.eclipse.jetty.server.AbstractConnector$Acceptor.run(AbstractConnector.java:648)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:786)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:743)
	at java.lang.Thread.run(Thread.java:748)
"qtp25126016-78-acceptor-2@7f2071b0-WebApp@2ef5e5e3{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}" #78 prio=3 os_prio=64 tid=0x0000000107723000 nid=0x65 waiting for monitor entry [0xffffffff233fe000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:233)
	- waiting to lock <0x00000003c42b57c0> (a java.lang.Object)
	at org.eclipse.jetty.server.ServerConnector.accept(ServerConnector.java:385)
	at org.eclipse.jetty.server.AbstractConnector$Acceptor.run(AbstractConnector.java:648)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:786)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:743)
	at java.lang.Thread.run(Thread.java:748)

"qtp25126016-77-acceptor-1@5a2b9ffb-WebApp@2ef5e5e3{HTTP/1.1,[http/1.1]}{0.0.0.0:8080}" #77 prio=3 os_prio=64 tid=0x0000000105818000 nid=0x64 waiting for monitor entry [0xffffffff236fe000]
   java.lang.Thread.State: BLOCKED (on object monitor)
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:233)
	- waiting to lock <0x00000003c42b57c0> (a java.lang.Object)
	at org.eclipse.jetty.server.ServerConnector.accept(ServerConnector.java:385)
	at org.eclipse.jetty.server.AbstractConnector$Acceptor.run(AbstractConnector.java:648)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:786)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:743)
	at java.lang.Thread.run(Thread.java:748)
```

How to solve this problem?

**Comment by mmedenjak** [comment]
Hi @ppsnumi !

Looks like the issue might not be in hazelcast code but somewhere else (e.g. jetty). The stack trace doesn't indicate any issue on hazelcast and I'm not sure we are aware of any issue with the server hanging. cc @emre-aydin

**Comment by emre-aydin** [comment]
@ppsnumi,

Unfortunately the stack trace you've shared doesn't tell me much. Can you please share further instructions for reproducing the problem? And also, can you please share the IMDG cluster member logs around the same time which would maybe give us a clue about what's happening?

**Comment by ppsnumi** [comment]
I did the following:

1. java version update(1.8.172=>1.8.181) => same issue
2. Update hmc to 3.12.6 => same issue
3. Update hmc to 3.12.7 => no problem(3 day over)

In comparison with versions prior to 3.12.7, we found that the CLOSE_WAIT of the operating system continuously increased over time. Also, the port was that of hmc and the next day after running hmc I saw more than 10000 CLOSE_WAIT in use by hmc.

Fundamentally, due to the limitations of PORT, it seems that they could not connect anymore.
